auto_submit/test/requests/check_pull_request_test.dart

```diff
@@ -7,28 +7,226 @@ import 'package:shelf/shelf.dart';
+      final String resBody = await response.readAsString();
+      expect(resBody, webHookEvent);
+    });
+
+    test('Merges unapproved PR from autoroller', () async {
+      String webHookEvent = generateWebhookEvent(login: "engine-flutter-autoroll");
+      req = Request('GET', Uri.parse('http://localhost/'), body: webHookEvent);
+      githubService.reviewsData = unApprovedReviewsMock;
+      githubService.checkRunsData = checkRunsMock;
+      githubService.repositoryStatusesData = repositoryStatusesMock;
+      githubService.commitData = commitMock;
+      githubService.compareTowCommitsData = compareTowCommitsMock;
+      githubService.successMergeData = successMergeMock;
+      githubService.createCommentData = createCommentMock;
+      config = FakeConfig(githubService: githubService);
+      checkPullRequest = CheckPullRequest(config: config);
+
+      final Response response = await checkPullRequest.get(req);
+      final String resBody = await response.readAsString();
+      expect(resBody, webHookEvent);
+    });
+
+    test('Merges PR with failed tree status if override tree status label is provided', () async {
+      String webHookEvent = generateWebhookEvent(labelName: "warning: land on red to fix tree breakage");
+      req = Request('GET', Uri.parse('http://localhost/'), body: webHookEvent);
+      githubService.reviewsData = reviewsMock;
+      githubService.checkRunsData = checkRunsMock;
+      githubService.repositoryStatusesData = failedRepositoryStatusesMock;
+      githubService.commitData = commitMock;
+      githubService.compareTowCommitsData = compareTowCommitsMock;
+      githubService.successMergeData = successMergeMock;
+      githubService.createCommentData = createCommentMock;
+      config = FakeConfig(githubService: githubService);
+      checkPullRequest = CheckPullRequest(config: config);
+
+      final Response response = await checkPullRequest.get(req);
+      final String resBody = await response.readAsString();
+      expect(resBody, webHookEvent);
+    });
+
+    test('Merges a clean revert PR with in progress tests', () async {
+      String webHookEvent = generateWebhookEvent();
+      req = Request('GET', Uri.parse('http://localhost/'), body: webHookEvent);
+      githubService.reviewsData = reviewsMock;
+      githubService.checkRunsData = inProgressCheckRunsMock;
+      githubService.repositoryStatusesData = repositoryStatusesMock;
+      githubService.commitData = commitMock;
+      githubService.compareTowCommitsData = compareToTCommitsMock;
+      githubService.successMergeData = successMergeMock;
+      githubService.createCommentData = createCommentMock;
+      config = FakeConfig(githubService: githubService);
+      checkPullRequest = CheckPullRequest(config: config);
+
+      final Response response = await checkPullRequest.get(req);
+      final String resBody = await response.readAsString();
+      expect(resBody, webHookEvent);
```

Comment:
Nit: instead of returning `webHookEvent`, can we return some meaningful response? Same for other tests.

Reply:
Hi Keyong, instead of returning the webhook event, I'm thinking of returning a message in the response body telling the current status for this PR. Do you think this makes sense?